stubs/docutils/docutils/statemachine.pyi

```diff
@@ -1,3 +1,194 @@
-from _typeshed import Incomplete
+from collections.abc import Callable, Generator, Iterator, Sequence
+from re import Match, Pattern
+from typing import Any, Generic, TypeVar, overload
+from typing_extensions import Self, TypeAlias
 
-def __getattr__(name: str) -> Incomplete: ...
+_TransitionResult: TypeAlias = tuple[list[str], str | None, list[str]]
+_TransitionMethod: TypeAlias = Callable[[Match[str], list[str], str], _TransitionResult]
+
+class StateMachine:
+    input_lines: StringList | None
+    input_offset: int
+    line: str | None
+    line_offset: int
+    debug: bool
+    initial_state: str
+    current_state: str
+    states: dict[str, State]
+    observers: list[Callable[[Self], None]]
+    def __init__(self, state_classes: Sequence[type[State]], initial_state: str, debug: bool = False) -> None: ...
+    def unlink(self) -> None: ...
+    def run(
+        self,
+        input_lines: list[str] | StringList,
+        input_offset: int = 0,
+        context: list[str] | None = None,
+        input_source: str | None = None,
+        initial_state: str | None = None,
+    ) -> list[str]: ...
+    def get_state(self, next_state: str | None = None) -> State: ...
+    def next_line(self, n: int = 1) -> str: ...
+    def is_next_line_blank(self) -> bool: ...
+    def at_eof(self) -> bool: ...
+    def at_bof(self) -> bool: ...
+    def previous_line(self, n: int = 1) -> str | None: ...
+    def goto_line(self, line_offset: int) -> str | None: ...
+    def get_source(self, line_offset: int) -> str: ...
+    def abs_line_offset(self) -> int: ...
+    def abs_line_number(self) -> int: ...
+    def get_source_and_line(self, lineno: int | None = None) -> tuple[str, int] | tuple[None, None]: ...
+    def insert_input(self, input_lines: list[str] | StringList, source: str) -> None: ...
+    def get_text_block(self, flush_left: bool = False) -> StringList: ...
+    def check_line(self, context: list[str], state: State, transitions: list[str] | None = ...) -> _TransitionResult: ...
+    def add_state(self, state_class: type[State]) -> None: ...
+    def add_states(self, state_classes: Sequence[type[State]]) -> None: ...
+    def runtime_init(self) -> None: ...
+    def error(self) -> None: ...
+    def attach_observer(self, observer) -> None: ...
+    def detach_observer(self, observer) -> None: ...
+    def notify_observers(self) -> None: ...
+
+class State:
+    patterns: dict[str, str | Pattern[str]] | None
+    initial_transitions: Sequence[str] | Sequence[tuple[str, str]] | None
+    nested_sm: type[StateMachine] | None
+    nested_sm_kwargs: dict[str, Any] | None
```

Comment:
Considering that these are always filled with not-None values in `__init__`, I'd skip the `| None` part. Might need a stubtest override.